src/Control/Monad/Error/Hoist.hs

```diff
@@ -41,6 +42,12 @@ instance MonadError e' m ⇒ HoistError m (Either e) e e' where
 instance (m ~ n, MonadError e' m) ⇒ HoistError m (EitherT e n) e e' where
   hoistError f = eitherT (throwError . f) return
 
+instance MonadError e' m ⇒ HoistError m (ExceptT e Identity) e e' where
+  hoistError f = either (throwError . f) return . runExcept
+
+instance MonadError e' m ⇒ HoistError m (ExceptT e m) e e' where
```

Comment:
Hmm, interesting! I suppose you get the same result for the existing `EitherT` one? Maybe we should consider changing the `EitherT` one to be like you have written the `ExceptT` one (assuming that it doesn't break inference in some places).

Honestly, I cannot really remember anymore what drove me to use two separate type variables in the first place (I thought it was either getting better inference, or avoiding some dangerous language extension.

In any case, I think we should be consistent; so if using just `m` seems to work, let's do it for `EitherT` as well (and verify that this doesn't break out existing use-cases).